kubectl-cloudflow/docker/client.go

```diff
@@ -136,3 +137,56 @@ func imageMatchesNameAndTag(image types.ImageSummary, imageNameAndTag string) bo
 	}
 	return false
 }
+
+// GetCloudflowStreamletDescriptorsForImage extracts the configuration of a Cloudflow label from a docker image
+// and the image with digest in a struct
+func GetCloudflowStreamletDescriptorsForImage(cli *client.Client, imageName string) (domain.CloudflowStreamletDescriptorsDigestPair, string) {
+
+	_, pullError := PullImage(cli, imageName)
+	if pullError != nil {
+		util.LogAndExit("Failed to pull image %s: %s", imageName, pullError.Error())
+	}
+
+	images, err := cli.ImageList(context.Background(), types.ImageListOptions{})
+
+	streamletDescriptorsLabelName := "com.lightbend.cloudflow.streamlet-descriptors"
+
+	if err != nil {
+		util.LogAndExit("Failed to list local docker images, %s", err.Error())
+	}
+
+	var descriptorsDigest domain.CloudflowStreamletDescriptorsDigestPair
+	for _, image := range images {
+		if len(image.RepoDigests) == 0 || len(image.RepoTags) == 0 || !imageMatchesNameAndTag(image, imageName) {
+			// not the image we are looking for
+			continue
+		}
+
+		// use the compressed streamlet descriptors base 64 value
+		raw := image.Labels[streamletDescriptorsLabelName]
+
+		// compressed data
+		compressed := base64.NewDecoder(base64.StdEncoding, bytes.NewReader([]byte(raw)))
+		reader, err := zlib.NewReader(compressed)
+		if err != nil {
+			util.LogAndExit("Failed to decompress the application descriptor, %s", err.Error())
```

Comment:
I think we should not log and exit from functions at this level.

Reply:
If we cannot decompress an image label there then verification has failed and we can fail fast. An reason not to?

Reply:
BTW this is the wrong error message, we're trying to read the streamlet descriptors here.